WordPress/gutenberg
# Add back the pattern overrides checkbox

## What?

Part of #53705, continued from #59268.

## Why?

There are some ongoing discussions about adding the "Allow instance overrides" checkbox back. This PR is created to explore this idea in action.

## How?

The checkbox will trigger the block rename modal if the block doesn't have a name already. This is still a confusing UX though.

## Testing Instructions




1. Create a synced pattern.
2. Add some supported blocks (`heading`, `paragraph`, `image`, and `button`) to the patterns and give them block names (in the editor settings -> Advanced panel)
3. Once they have block names, checking "Allow instance overrides" checkbox should not prompt the modal.
4. Otherwise, if the block doesn't have a name, checking the checkbox should prompt a block renaming modal.
5. Insert the pattern in a post and edit those blocks.
6. The blocks should be overridable and show on the frontend.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/7753001/298ce053-3230-4284-9c3d-afd4474c27ca
